tensorflow/python/eager/backprop.py

```diff
@@ -923,6 +926,11 @@ def gradient(self,
     if self._tape is None:
       raise RuntimeError("GradientTape.gradient can only be called once on "
                          "non-persistent tapes.")
+    for v in target:
```

Comment:
A similar check in gradient is being added in #27183.

Reply:
This line with "for v in target" fails if target is a scalar tensor or if it's a nested structure. I think there's a nest.flatten(target) somewhere else in this function; can you iterate over that instead?